src/kvstore/raftex/Host.h

```diff
@@ -244,6 +251,12 @@ class Host final : public std::enable_shared_from_this<Host> {
 
   mutable std::mutex lock_;
 
+  // If stopped_ is true, we will not send any request to the peer;
+  // If stopped_ is false:
+  //  1. no mater whether paused_ is true or not, we can send HB request or AskForVote request;
+  //  2. Only if paused_ is false, we can send appendlog request, of course, including HB
+  //     request and AskForRequest request
+  // See canAppendLog() and canSendHBOrVote()
```

Comment:
I suppose you are talking about "Why not just use stopped_ xxxx", not paused_. Correct?

Reply:
Sorry I didn't say clearly. 
At the first time, `stopped_` is planned to control whether the `Host` could send AppendLogs or Heartbeat to peers, which is the `canAppendLog`. On the other hand , `paused` is not used actually.

In this pull request:
* `stopped` controls whether AppendLogs/Heartbeat and AskForVote is allowed ()
* `paused` controls whether Heartbeat/AskForVote is allowed

If what I said is correct, I suggest to:
1. Maybe rename `canAppendLog` as `canSendRequest` (or something similar). And only check `stopped_` in this function.
2. In `canSendHBOrVote`, check whether peer host is down or not by `paused_`. Use this function only in Heartbeat. (Because in AskForVote, we only check `stopped_` actually, so maybe rename it as canSendHB?)

Reply:
Only when both stopped_ and paused_ are false, we can send AppendLog request, this is the same logic as before; as long as stopped_ is false, we can send AskForVote request, this is also the same logic as before; The only difference is for Heartbeat request. Previously the condition for sending Heartbeat is the same as AppendLog, but in this PR, it is the same as AskForVote. The intention is to use HB request to detect whether the peer is back online so we need to send HB even if the peer was down.

Reply:
Yeah, I got your point. I mean we don't always need to keep same logic as before. It is a not quite intuitive to me. That's OK.

Reply:
The intention is to not bring breaking changes, so I make as less change as possible with the current code.